src/ext/java/org/opentripplanner/ext/fares/model/FareLegRule.java

```diff
@@ -8,20 +8,15 @@ public record FareLegRule(
   @Nullable String legGroupId,
   @Nullable String networkId,
   @Nullable String fromAreaId,
-  @Nullable String toAreadId,
+  @Nullable String toAreaId,
   @Nullable FareDistance fareDistance,
   @Nonnull FareProduct fareProduct
 ) {
-  public FareLegRule(
-    String legGroupId,
-    String networkId,
-    String fromAreaId,
-    String toAreaId,
-    FareProduct fareProduct
-  ) {
-    this(legGroupId, networkId, fromAreaId, toAreaId, null, fareProduct);
-  }
   public String feedId() {
     return fareProduct.id().getFeedId();
   }
+
+  public static FareLegRuleBuilder of(FareProduct fp) {
```

Comment:
This will need to ingest an arbitrary number of FareProducts, it might make sense to have this method use varargs and overload it with another that accepts an `ArrayList`? I like the change to use the builder though!

Reply:
It makes total sense but I would like to ask you to do that in your PR.